qiskit/circuit/library/arithmetic/multipliers/hrs_cumulative_multiplier.py

```diff
@@ -0,0 +1,135 @@
+               │        ││        ││        │
+        out_4: ┤        ├┤6       ├┤5       ├
+               │        ││        ││        │
+        out_5: ┤        ├┤        ├┤6       ├
+               │        ││        ││        │
+        aux_0: ┤7       ├┤7       ├┤7       ├
+               └────────┘└────────┘└────────┘
+
+    Multiplication in this circuit is implemented in a classical approach by performing
+    a series of shifted additions using one of the input registers while the qubits
+    from the other input register act as control qubits for the adders.
+
+    **References:**
+
+    [1] Häner et al., Optimizing Quantum Circuits for Arithmetic, 2018.
+    `arXiv:1805.12445 <https://arxiv.org/pdf/1805.12445.pdf>`_
+
+    """
+
+    def __init__(
+        self,
+        num_state_qubits: int,
+        num_result_qubits: Optional[int] = None,
+        adder: Optional[QuantumCircuit] = None,
+        name: str = "HRSCumulativeMultiplier",
+    ) -> None:
+        r"""
+        Args:
+            num_state_qubits: The number of qubits in either input register for
+                state :math:`|a\rangle` or :math:`|b\rangle`. The two input
+                registers must have the same number of qubits.
+            num_result_qubits: The number of result qubits to limit the output to.
+                If number of result qubits is `n`, modulo :math:`2^n` is performed
+                to limit the output to the specified number of qubits. Default
+                value is ``2 * num_state_qubits``.
+            adder: Half adder circuit to be used for performing multiplication. The
+                CDKMRippleCarryAdder is used as default if no adder is provided.
+            name: The name of the circuit object.
+        Raises:
+            ValueError: If ``num_result_qubits`` is not default and a custom adder is provided.
+        """
+        if num_result_qubits is None:
+            num_result_qubits = 2 * num_state_qubits
+
+        if num_result_qubits != 2 * num_state_qubits and adder is not None:
+            raise ValueError("Only default adder is supported for modular multiplication.")
+
+        super().__init__(num_state_qubits, num_result_qubits, name=name)
+
+        # define the registers
+        qr_a = QuantumRegister(num_state_qubits, name="a")
+        qr_b = QuantumRegister(num_state_qubits, name="b")
+        qr_out = QuantumRegister(num_result_qubits, name="out")
+        self.add_register(qr_a, qr_b, qr_out)
+
+        # prepare adder as controlled gate
+        if adder is None:
+            from qiskit.circuit.library import CDKMRippleCarryAdder
+
+            adder = CDKMRippleCarryAdder(num_state_qubits, kind="half")
+
+        # get the number of helper qubits needed
+        num_helper_qubits = adder.num_ancillas
+
+        # add helper qubits if required
+        if num_helper_qubits > 0:
+            qr_h = AncillaRegister(num_helper_qubits, name="aux")  # helper/ancilla qubits
```

Comment:
Could you name this helper (or help?) to be consistent with the adders?
```suggestion
            qr_h = AncillaRegister(num_helper_qubits, name="helper")  # helper/ancilla qubits
```

Reply:
Thanks for the suggestion :)